feat: add mock market banner dev setting for UI testing

limichange opened the pull request:
## Summary
- Add `enableMockMarketBanner` option in dev settings to enable mock market banner data
- Add mock data for market banner list in ServiceMarketV2 for UI testing
- Add UI toggle in dev settings section for enabling mock banner
- Refactor Home page and token list components
- Simplify Market mobile layout

## Test plan
- [ ] Enable mock market banner in dev settings
- [ ] Verify banner displays with mock data on Market page
- [ ] Verify toggle works correctly
- [ ] Test on mobile and desktop platforms

limichange committed:
feat: add mock market banner dev setting for UI testing

- Add enableMockMarketBanner option in dev settings
- Add mock data for market banner list in ServiceMarketV2
- Add UI toggle in dev settings section for enabling mock banner
- Refactor Home page and token list components
- Simplify Market mobile layout

revan-zhang commented:
### :white_check_mark: **Snyk checks have passed. No issues have been found so far.**

| Status | Scanner | ![](https://res.cloudinary.com/snyk/image/upload/r-d/scm-platform/snyk-pull-requests/severity-critical.svg) Critical | ![](https://res.cloudinary.com/snyk/image/upload/r-d/scm-platform/snyk-pull-requests/severity-high.svg) High | ![](https://res.cloudinary.com/snyk/image/upload/r-d/scm-platform/snyk-pull-requests/severity-medium.svg) Medium | ![](https://res.cloudinary.com/snyk/image/upload/r-d/scm-platform/snyk-pull-requests/severity-low.svg) Low | Total (0) |
|---|---|---|---|---|---|---|
| :white_check_mark: | Open Source Security| 0 | 0 | 0 | 0 | [ 0 issues](https://app.snyk.io/org/rayston92/pr-checks/4861fe53-2e39-41e0-8dda-7095a5a72df6?refs=7e249979b127831fbd668e0c2af05e5502068539&source=prChecksComment) |
| :white_check_mark: | Licenses| 0 | 0 | 0 | 0 | [ 0 issues](https://app.snyk.io/org/rayston92/pr-checks/4861fe53-2e39-41e0-8dda-7095a5a72df6/license?refs=7e249979b127831fbd668e0c2af05e5502068539&source=prChecksComment) |

:computer: Catch issues earlier using the plugins for [VS Code](https://marketplace.visualstudio.com/items?itemName=snyk-security.snyk-vulnerability-scanner), [JetBrains IDEs](https://plugins.jetbrains.com/plugin/10972-snyk-security), [Visual Studio](https://marketplace.visualstudio.com/items?itemName=snyk-security.snyk-vulnerability-scanner-vs-2022), and [Eclipse](https://marketplace.eclipse.org/content/snyk-security#external-install-button).

originalix commented:
@codex review

limichange committed:
fix: add type annotation for banner list items in MarketBannerList component

- Added IMarketBannerItem type to the banner list mapping for better type safety and clarity in the MarketBannerList component.

limichange committed:
refactor: remove type annotation for banner items in MarketBannerList component

- Simplified the mapping of banner items by removing the explicit IMarketBannerItem type annotation in the MarketBannerList component for improved readability.
- Updated the useMarketBannerList hook to return a more generic bannerList type.

limichange committed:
fix: invalidate market banner cache when mock setting changes

- Add clearMarketBannerCache method in ServiceMarketV2
- Add RefreshMarketBannerList event to event bus
- Update useMarketBannerList hook to depend on enableMockMarketBanner
- Clear banner cache when mock setting is toggled in dev settings

limichange committed:
fix: ensure isLoading is boolean type

limichange committed:
refactor: enhance type safety and return type for market banner methods

- Added IMarketBannerItem type annotation to MOCK_MARKET_BANNER_LIST for improved type safety.
- Updated fetchMarketBannerList method to return a Promise of IMarketBannerItem array.
- Modified clearMarketBannerCache method to return a Promise<void> for consistency.

limichange committed:
refactor: clean up market banner related code and improve comments

- Removed unused RefreshMarketBannerList event from event bus.
- Updated the subtitle for the mock market banner setting to English for consistency.
- Enhanced comments in useMarketBannerList and clearMarketBannerCache methods for better clarity on functionality.

limichange committed:
Merge branch 'x' into feat/add-mock-market-banner-dev-setting

limichange committed:
Merge branch 'x' into feat/add-mock-market-banner-dev-setting

originalix commented on packages/kit-bg/src/services/ServiceMarketV2.ts:
```diff
@@ -32,9 +33,127 @@ import type {
 
 import ServiceBase from './ServiceBase';
 
+const MOCK_MARKET_BANNER_LIST: IMarketBannerItem[] = [
```
拆到独立的 constant 文件里吧

limichange reviewed (commented):
已修复，将 `MOCK_MARKET_BANNER_LIST` 拆分到独立的 `ServiceMarketV2.const.ts` 文件中。放在同目录下比较合适，方便查找和维护。

limichange commented:
已修复，将 `MOCK_MARKET_BANNER_LIST` 拆分到独立的 `ServiceMarketV2.const.ts` 文件中。放在同目录下比较合适，方便查找和维护。

limichange committed:
refactor: extract MOCK_MARKET_BANNER_LIST to separate constant file

Move mock market banner data to ServiceMarketV2.const.ts for better code organization

limichange committed:
Merge branch 'x' into feat/add-mock-market-banner-dev-setting

ByteZhang1024 committed:
Merge branch 'x' into feat/add-mock-market-banner-dev-setting
